### DISCUSSION TITLE
Missing Drive but shown on TeslaFi

### DISCUSSION THREAD

**Discussion by Brain1000** [teslamate-org/teslamate#1312] [Q&A]
**Describe the bug**

I made a short 13 minute drive to the shops which was recorded fine.  After 45 minutes I returned home, but the return journey was not recorded.

As a very new convert to TeslaMate I still have access to TeslaFi and this recorded the return drive fine.

I note some errors in the logs but TeslaFi somehow seems to overcome the problem.

**Expected behavior**

To record the drive as TeslaFi does.

**Relevant entries from the logs**

```
[36mteslamate_1  |[0m 2020-06-04T01:22:11.701140612Z 02:22:11.700 car_id=1 [info] Fetching vehicle state ...
[36mteslamate_1  |[0m 2020-06-04T01:22:12.602932997Z 02:22:12.602 car_id=1 [info] Start / :asleep
[36mteslamate_1  |[0m 2020-06-04T01:22:12.607553277Z 02:22:12.607 car_id=1 [info] Disconnecting ...
[36mteslamate_1  |[0m 2020-06-04T18:08:54.035391140Z 19:08:54.035 car_id=1 [info] Start / :online
[36mteslamate_1  |[0m 2020-06-04T18:08:54.044296486Z 19:08:54.044 car_id=1 [info] Connecting ...
[36mteslamate_1  |[0m 2020-06-04T18:08:54.046501717Z 19:08:54.045 car_id=1 [info] Driving / Start
[36mteslamate_1  |[0m 2020-06-04T18:23:02.319673646Z 19:23:02.319 car_id=1 [info] Driving / Ended / 15 km – 14 min
[36mteslamate_1  |[0m 2020-06-04T18:23:02.319903300Z 19:23:02.319 car_id=1 [info] Start / :online
[36mteslamate_1  |[0m 2020-06-04T18:26:28.092466273Z 19:26:28.092 car_id=1 [info] Suspending logging
[36mteslamate_1  |[0m 2020-06-04T19:26:23.898181820Z 20:26:23.897 car_id=1 [info] Fetching vehicle state ...
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897013517Z 20:26:44.896 [error] TeslaApi.Error / 408 – %{
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897053407Z   "error" => "vehicle unavailable: {:error=>\"{\\\"error\\\": \\\"timeout\\\"}\"}",
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897061343Z   "error_description" => "",
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897066458Z   "response" => nil
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897071528Z }
[36mteslamate_1  |[0m 2020-06-04T19:26:44.897295043Z 20:26:44.896 car_id=1 [error] Error / :vehicle_unavailable
[36mteslamate_1  |[0m 2020-06-04T19:26:57.919049836Z 20:26:57.918 [info] Stream.Timeout / 5
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639515988Z 20:27:26.639 [error] TeslaApi.Error / 408 – %{
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639556109Z   "error" => "vehicle unavailable: {:error=>\"{\\\"error\\\": \\\"timeout\\\"}\"}",
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639564366Z   "error_description" => "",
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639569769Z   "response" => nil
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639574856Z }
[36mteslamate_1  |[0m 2020-06-04T19:27:26.639877039Z 20:27:26.639 car_id=1 [error] Error / :vehicle_unavailable
[36mteslamate_1  |[0m 2020-06-04T19:27:29.913940063Z 20:27:29.913 [info] Stream.Timeout / 6
[36mteslamate_1  |[0m 2020-06-04T19:28:26.937928739Z 20:28:26.937 [info] Stream.Timeout / 7
[36mteslamate_1  |[0m 2020-06-04T22:09:01.946460001Z 23:09:01.945 [info] Stream.Timeout / 8
[36mteslamate_1  |[0m 2020-06-04T22:09:03.124030180Z 23:09:03.123 [error] Vehicle Error: Vehicle is offline
[36mteslamate_1  |[0m 2020-06-04T22:09:33.125006526Z 23:09:33.124 [info] Stream.Timeout / 9
[36mteslamate_1  |[0m 2020-06-04T22:09:34.232816068Z 23:09:34.232 [error] Vehicle Error: Vehicle is offline
```

**Environment**

- TeslaMate version: 1.19.3
- Type of installation: Docker
- OS TeslaMate is installed on: Ubuntu 18.04

**Comment by adriankumpf** [comment]
I can't really explain it. TeslaFi uses exactly the same API as we do (except for the Streaming API). I don't know what part of Tesla's backend caused the car to be reported as "unavailable". Maybe it has something to do with the fact that several loggers have just accessed the car or that TeslaFi has communicated with another server. But who knows – only Tesla can answer that.

**Comment by bpankz** [comment]
noticing some of the same behavior here as well. only using teslamate, no other logger. Multiple drives have missed up to 20 minutes of data, about 1-2 times per week, not really a pattern I can make of it. I'll dig a bit deeper and see what I can find but it happened this morning as well, missed about 5 minutes even though the car was on and running ac for a good 5 minutes while I was loading up the car

**Comment by SDekkers** [comment]
I'm seeing the same; a few short drives this morning are missing from the grafana drives list, while another tool has found them fully. I'll check the logs tonight at home to see if there's anything related publishable. Perhaps it's the streamingAPI having issues at Tesla.

Edit: I'm also noticing the TeslaMate interface is unavailable from time to time.

**Comment by cwanja** [comment]
Anyone relating this to the 20.12 update that included “GPS fixes”? I got the update two days ago, never drove, and then the first time I did drive, TeslaMate missed the two drives which were about 30 minutes total.

**Comment by Brain1000** [comment]
@cwanja 
I have been on 20.12.1 for 11 days (on a S75D in the UK) and done 12 drives but with only the one problem mentioned above - the rest were all ok.

**Comment by lux4rd0** [comment]
Just a little bit on my own - I've seen several of my drives now either not show up - not miss the start. I specifically have now started to access my mobile app when I start my drive to validate that Tesla Mothership is seeing my activity. Looking at the logs, I get:

```
12:20:07.817 [info] Stream.Timeout / 321
12:20:38.289 [info] Stream.Timeout / 322
12:21:08.713 [info] Stream.Timeout / 323
12:21:39.196 [info] Stream.Timeout / 324
12:22:09.623 [info] Stream.Timeout / 325
12:22:12.507 car_id=1 [info] Driving / Start
```

That missed about 2-3 minutes of my drive.

Also several of these:

` [warn] Discarded incomplete fetch result:`

Not much we can do about that. I will say that before the switch to the stream format, I don't recall missing any of my drives but I can't be 100% certain of it. Will switch off the stream configuration and see if that helps.

**Comment by Brain1000** [comment]
I wonder if it would be possible to change the behaviour such that if the streaming was timing out then to automatically revert to standard polling for a period of time?

**Comment by lux4rd0** [comment]
Took a handful of drives again today - missing almost 6-10 minutes of the start of  each of the drives. What's the best way to help debug this? I see my car connected via the Tesla mobile app. I see everything in real time - but it's not getting logged by Teslamate. I've disabled the streaming API and it didn't seem to matter.

```
19:32:22.902 [error] TeslaApi.Error / 408 – %{
  "error" => "vehicle unavailable: {:error=>\"{\\\"error\\\": \\\"timeout\\\"}\"}",
  "error_description" => "",
  "response" => nil
}
19:32:22.902 car_id=1 [error] Error / :vehicle_unavailable
20:21:35.810 car_id=1 [error] Error / :timeout
20:22:08.844 car_id=1 [info] Driving / Start
20:26:21.385 [error] TeslaApi.Error / 408 – %{
  "error" => "vehicle unavailable: {:error=>\"{\\\"error\\\": \\\"timeout\\\"}\"}",
  "error_description" => "",
  "response" => nil
}
20:26:21.837 car_id=1 [warn] Discarded incomplete fetch result: %TeslaApi.Vehicle{api_version: 8, backseat_token: nil, backseat_token_updated_at: nil, calendar_enabled: true, charge_state: XXXXX(Removed)}
20:30:35.063 car_id=1 [info] Driving / Ended / 9 km – 8 min
20:30:35.063 car_id=1 [info] Start / :online
20:31:25.195 car_id=1 [info] Driving / Start
20:37:20.287 car_id=1 [info] Driving / Ended / 5 km – 6 min
20:37:20.287 car_id=1 [info] Start / :online
20:53:15.520 car_id=1 [info] Suspending logging
21:53:26.942 car_id=1 [info] Driving / Start
21:59:51.757 car_id=1 [info] Driving / Ended / 1 km – 6 min
21:59:51.757 car_id=1 [info] Start / :online
```

Those items at then end are where I missed 6-10 minutes of drive time.

**Comment by remusache** [comment]
Same here…
Switched from TeslaFi to TeslaMate only to discover that TeslaMate is not 100% logging the drives.
Missing almost 6-10 minutes of the start of each of the drives. If not more sometimes.

**Comment by tobang** [comment]
I'm seeing the same behaviour. Today, it missed an entire drive, with Stream.Timeout in the log. I am still using TeslaFi and here the drive is showing up. Let me know if there is something I can do to help debug the issue.

**Comment by cwanja** [comment]
Has anyone checked to see if TeslaMate was "attempting to fall asleep" at the time that the missing drive happened? I know I have missed a start of a charging session or a drive due to the car randomly waking up and then attempting to fall asleep for extended amount of hours and I miss the start of or an entire drive (pending how short it was) due to this.

**Comment by EasternPA** [comment]
It is so rare that TeslaMate misses a drive, but of course one of the few times was when my garage took the car out for the road test during inspection. He disappeared down the road and I knew I was going to see what he did later on. I checked and the drive was missing! Made me think he knew to click something in the menus to make it stop sending data to Tesla.

**Comment by johnpane** [comment]
I just moved from TeslaFi to TeslaMate and am surprised with the truncation of the beginning of drives. In four years I never saw this from TeslaFi yet it is happening daily with TeslaMate. I have the car set to "always connected." (2017 Model S with sw 2020.48.37.6.)

**Reply by johnpane** [reply]
I just took a close look at data from TeslaMate and TeslaFi. Both are showing approximately the same GPS track. The difference is that TeslaFi is detecting that there is missing data at the end and is identifying the start point of the drive as the location the car was last parked. So, there is a gap between that parking spot and the beginning of the GPS track but importantly, the geofence locations are correctly identified. 

My conclusion is that this is not a data collection issue but an interpretation issue. It seems like it might be easily solvable: when a drive starts use the most recent parking location as the "start" of the drive, regardless where the gps track begins.

**Reply by DrMichael** [reply]
> My conclusion is that this is not a data collection issue but an interpretation issue. It seems like it might be easily solvable: when a drive starts use the most recent parking location as the "start" of the drive, regardless where the gps track begins.

I think, we had that discussion already. Conclusion was not to do that.
